Automattic/newspack-plugin
# newspack-components library

This PR sets up the ability to publish our components to npm as a stand-alone package. This work is largely based on the two Calypso PRs linked below. Thanks to @ockham for pointing me in the right direction!

## To test

1. In this repo, `cd assets/src/components; npm install;`.
2. `npm publish --dry-run` to see what would be published.
3. In a separate repo, [get the latest version of the npm package](https://www.npmjs.com/package/newspack-components) (0.1.5+)
4. Try something like this to verify the components can be used in third-party applications:
```
import { TextControl, Button, Card } from 'newspack-components';
import { Component, render } from '@wordpress/element';

class PackageTest extends Component {
	render() {
		return (
			<Card>
				<TextControl
					label={'Text Input' }
					value=''
					onChange={ () => {} }
				/>
				<Button isPrimary >
					Button
				</Button>
			</Card>
		);
	}
}

render( <PackageTest />, document.getElementById( 'package-test' ) );
```

You should see something like this:

<img width="534" alt="Screen Shot 2019-05-24 at 11 21 28 AM" src="https://user-images.githubusercontent.com/7317227/58348677-22d0cb80-7e16-11e9-91fe-d062ae062ead.png">
